runtime/lua/vim/filetype.lua

```diff
@@ -2376,8 +2361,28 @@ local function dispatch(ft, path, bufnr, ...)
   end
 end
 
+-- Caches calls to string.find
```

Comment:
Expand on this comment a bit more, it is not at all obvious what this is for if you are just reading for the first time

Suggestion:
```suggestion
-- Lookup table/cache for patterns that contain an environment variable pattern, e.g. ${SOME_VAR}.
```